new/resources-anywhere.md

```diff
@@ -0,0 +1,211 @@
+
+```ruby
+class MyResource < Chef::Resource
+  ...
+
+  class Provider < Chef::Provider
+    ...
+  end
+end
+```
+
+This has several nice benefits:
+- Supports the goal of resources outside of `Chef::Resource`.
+- Encourages the creation of providers in the same file as the resource.
+- An absolute minimum of extra typing when linking a resource to a provider.
+
+### Creating a non-DSL resource
+
+Some users create base resources and .  This can be accomplished by calling `chef_recipe_dsl false` in the class, which will undef the DSL for it:
+
+```ruby
+class SuperAwesomeBaseResource < Chef::Resource
+  chef_recipe_dsl false
+
+  ...
+end
+```
+
+### Creating a DSL resource with a different name
+
+Sometimes the name of your class isn't the name that you need.  In that case, you can change the DSL with `chef_recipe_dsl`.
+
+```ruby
+class SuperMountPointResource < Chef::Resource
+  chef_recipe_dsl name: 'mount_point'
+end
+```
+
+This will remove the `super_mount_point_resource` DSL and add `mount_point` as DSL.
+
+### Single File LWRPs
+
+Since HWRPs are now easily declared in a single file, it would be macabre of us not to allow single file LWRPs as well.  We propose two new syntaxes in LWRP resources to enable this:
+
+1. `action :create do ... end` (in the resource)
+   - Creates a Provider class extending LWRPBase if it does not already exist.
+   - Sets `use_inline_resources` to true if not explicitly set.
+   - Adds `:create` to `allowed_actions` on the resource.
+   - Calls `action :create do ... end` on the Provider class.
+
+NOTE: this does not address `load_current_resource`, as there are other proposals brewing that may address that in other ways.  Users can open up the provider class with `class Provider < Chef::Provider::LWRPBase` if they need to implement this.
+
+### Backwards Compatibility and Deprecation
+
+Backwards compatibility is king.
+
+- All things that currently work in Chef 12 must continue to work in Chef 12.
+- 99.99% of things that currently work in Chef 12, must continue to work in Chef 13.
+- As much as possible should work in 12.X, so that people in Chef 12 have a story when we move to 13 and so people can begin converting now.
+
+#### Accessing `Chef::Resource::File` or `Chef::Provider::File` directly
+
+In Chef 13, core resources are simply moving out of the Resource and Provider namespaces.  However, in Chef 12, there will be resources and providers who extend from them or construct them directly.
+
+In Chef 12, we will create classes in `Chef::Resource` extending from the real Resources which emit a deprecation warning on initialize or subclassing (`class Chef::Resource::File < Chef::Resources::File ...`).
```

Comment:
Still think this is a lot of work we're asking people to do, just in order to be able to get around the `::File` issue